### PR TITLE
enum conversion

### PR DESCRIPTION
**Pull request by tracylynne99**
Converted several classes to enums

### CODE REVIEW

File: src/main/java/org/jfree/chart/plot/SeriesRenderingOrder.java [hunk 1]
```diff
@@ -36,107 +36,27 @@
...
      */
-    public static final SeriesRenderingOrder FORWARD
-            = new SeriesRenderingOrder("SeriesRenderingOrder.FORWARD");
+    FORWARD,
 
     /**
      * Render series in the order N-1, N-2, ..., 2, 1, 0, where N is the
      * number of series.
      */
-    public static final SeriesRenderingOrder REVERSE
```
Comment (TheBoegl) [inline src/main/java/org/jfree/chart/plot/SeriesRenderingOrder.java]: and here

File: src/main/java/org/jfree/chart/renderer/AreaRendererEndType.java [hunk 2]
```diff
@@ -35,104 +35,29 @@
...
      * The area tapers from the first or last value down to zero.
      */
-    public static final AreaRendererEndType TAPER = new AreaRendererEndType(
-            "AreaRendererEndType.TAPER");
+    TAPER,
 
     /**
      * The area is truncated at the first or last value.
      */
-    public static final AreaRendererEndType TRUNCATE = new AreaRendererEndType(
```
Comment (TheBoegl) [inline src/main/java/org/jfree/chart/renderer/AreaRendererEndType.java]: here

File: src/main/java/org/jfree/chart/renderer/AreaRendererEndType.java [hunk 3]
```diff
@@ -35,104 +35,29 @@
...
-            "AreaRendererEndType.TRUNCATE");
-
-    /**
-     * The area is levelled at the first or last value.
-     */
-    public static final AreaRendererEndType LEVEL = new AreaRendererEndType(
```
Comment (TheBoegl) [inline src/main/java/org/jfree/chart/renderer/AreaRendererEndType.java]: and here

**Comment by TheBoegl** [comment]
I'm sorry I think I messed up my diff view and wrote comment, were you were right.

**Comment by simon04** [comment]
@tracylynne99: Thanks for modernizing the code base!

@jfree: Your dealing with pull requests is nontransparent and super frustrating! :angry: I opened the analogous PR #84 already in May 2018. To the day of writing, that one hasn't received a single comment. It's a wasteful use of sparse contributor's resources and is super demotivating for possible future contributions/contributors. Why would one contribute, if his/her contributions are simply ignored and made obsolete over time by other contributors redoing basically the same work?!

**Comment by tomas-pluskal** [comment]
@simon04 I think David just gave up on maintaining JFreeChart. Perhaps we should fork the project and form a group of interested developers who would maintain the fork and accept pull request (I don't know if anyone is willing to do more active development of the project). Opinions, ideas?

**Comment by tracylynne99** [comment]
@simon04 Oops, sorry, I didn't mean to step on your toes! It took forever for him to even accept my pull request, so I was pretty surprised that it actually happened.
@tomas-pluskal I think you are right.  I wonder if it's because he's paying more attention to jfreechart-fse? I haven't really looked at that in a while so I dont know.

I'd definitely like to do more active developing for this project!  Making use of Java's Generics capability wherever possible would be especially joyous.

**Comment by simon04** [comment]
@tomas-pluskal The question and my hope is that this project can gain some momentum without forking it, i.e., enlarging the maintainer team and keep the project named "jfreechart".
@tracylynne99 I'm not blaming you; it's impossible/hard to track all open PR before submitting.

Yes, generics, continuous integration (using Travis CI for instance), merging PR from GitHub, reviewing patches from Sourceforge, doing frequent releases (e.g., quarterly) … There's plenty of work to be done. And it does not have to last one a single person's shoulder.

**Comment by jfree** [comment]
Hi all, I do feel bad about not being able to commit the time to JFreeChart that it really requires.  I'm open to ideas about getting the project moving along better, and would welcome assistance.

**Comment by tomas-pluskal** [comment]
@jfree Why not open up the project and give commit rights to other developers who could then merge pull requests, publish releases, and overall support more active development? You could file an issue here with a small survey to see how many people would be interested.

**Comment by tracylynne99** [comment]
I would definitely be interested!

**Comment by jfree** [comment]
Let me think about whether to open things up completely.  In the meantime, if there is any pull request that (a) merges cleanly and (b) at least two of you (tracy, simon and tomas) agree should be merged, then I'll do it.

**Pull request merged**